.circleci/config.yml

```diff
@@ -9,26 +9,19 @@ jobs:
       - checkout
       - setup_remote_docker:
           reusable: true
-          exclusive: false
+          # run tests run on separate docker engines to avoid interference and
+          # and resource contention
+          exclusive: true
       - restore_cache:
           key: yarpc-go
       - run:
-          name: crossdock
-          command: make ci
-          environment:
-            - CI_TYPES: crossdock
-      - run:
-          name: 1.8 lint test examples
-          command: make ci
-          environment:
-            - CI_TYPES: deps lint test examples
-            - DOCKER_GO_VERSION: 1.8
-      - run:
-          name: 1.7 cover
-          command: make ci
-          environment:
-            - CI_TYPES: deps cover
-            - DOCKER_GO_VERSION: 1.7
+         name: run make ci
+         command:
+           case $CIRCLE_NODE_INDEX in
+           0) CI_TYPES=crossdock make ci ;;
+           1) DOCKER_GO_VERSION=1.8 CI_TYPES="deps lint test examples" make ci ;;
+           2) DOCKER_GO_VERSION=1.7 CI_TYPES="deps lint test examples" make ci ;;
```

Comment:
I think we'll want coverage on 1.8 now (since we just updated to the 1.8.x branch internally), and we definitely don't want lint on 1.7 (since `go vet` occasionally gives conflicting advice on different minor versions).